src/block/block_cache.c

```diff
@@ -343,8 +341,8 @@ __wt_blkcache_get_or_check(
      * If more than the configured fraction of all file objects is likely to fit in the OS buffer
      * cache, don't use this cache.
      */
-    if (blkcache->system_ram >=
-      __blkcache_estimate_filesize(session) * blkcache->fraction_in_os_cache) {
+    if (__blkcache_estimate_filesize(session) <
+      (blkcache->system_ram * 100) / blkcache->percent_file_in_os_cache) {
         WT_STAT_CONN_INCR(session, block_cache_bypass_get);
         return (WT_BLKCACHE_BYPASS);
     }
```

Comment:
For clarity, I would change this to: 

```
 if ((__blkcache_estimate_filesize(session) * blkcache->percent_file_in_os_cache) / 100 <
       blkcache->system_ram * 100))
```

This way the formula matches to how we explain this in the comment.